### DISCUSSION TITLE
NextJS - Incremental Static Regeneration on Vercel constantly times out (on a pro plan - 60s time-out limit)

### DISCUSSION THREAD

**Discussion by tompbsmith** [vercel/community#305] [Help]
I'm having issues with a NextJS project hosted on a Vercel pro plan. Routes that are set up with ISR just keep timing out. I have revalidate set to 120s currently but I've tried it with 60s, 30s, and 6000s too just in case I was calling it too often.

I'm using Strapi as the headless CMS that serves the API for NextJS to build pages from and is deployed on Render's German region. The database Strapi uses is a mongodb databse hosted on MongoDB Atlas and deployed on MongoDB's Ireland AWS region. (I've also set the Serverless Functions on Vercel to be served from London, UK but I'm not sure if that affects ISR?)

There are 3 different dynamic routes with about 20 pages each and on build-time they average 6999ms, 6508ms and 6174ms respectively. Yet at run-time, if I update some content in the Strapi CMS and wait the 120s that I've set for revalidate the page hardly ever gets rebuilt. If I look at the vercel dashboard "Functions" tab that shows realtime logs, I see that many of the pages have attempted to rebuild all at the same time and they are all hitting the 60s time-out limit.

I also have the vercel logs being sent to LogTail and if I filter logs for the name of the page that I've edited, I can see that it returns a 304 status code before 120s has passed as expected but then after 120s it tries to fetch and build the new page and nearly always returns the time-out error.

Here is a screenshot of my vercel realtime logs. As you can see, many of the pages are all trying to rebuild at once but I'm not sure why, I've only changed the data for one page in this instance.

![OOx7k](https://user-images.githubusercontent.com/2126789/152614932-cf5d0163-98a4-4c40-a60d-63007d38498b.png)

To try and debug the issue, I decided to create a Postman Flow for building one of the dynamic routes and then added up the time for each api call that is needed to build the page and I get 7621ms on average after a few tests. Here is a screenshot of the Postman console:

![1D2Gl](https://user-images.githubusercontent.com/2126789/152614988-134d93c2-9c84-4541-add5-c6c8dbe0f106.png)

Why are so many of them trying to rebuild at the same time if nothing has changed in the CMS for all of those pages but the 1 I've deliberately changed myself? And secondly, why at build time does it only take an average of 6000ms to build a page but during ISR they are hitting the 60s time-out limit?

Could it be because so many rebuilds are being triggered that they are all end up causing each other to time-out? If so, then how to I tackle that first issue?

I'm not that experienced with ISR on Vercel so I'm hoping I'm just doing something wrong or I've not got a setting correct on vercel etc. but after looking on stackoverflow and other websites, I believe I'm using ISR as expected. If anybody has any ideas or advice about what might be going on, I'd very much appreciate it.

**Comment by samcx** [comment]
Apologies for the late response!

Can you confirm if you are still experiencing this?

**Reply by haseeb-bewonder** [reply]
Hi @samsisle , I'm facing this issue and I'm using Wordpress as my headless cms. And the issue exact same as mentioned above.

Can you please guide me on this ?

**Reply by amyegan** [reply]
Hi @haseeb-bewonder! Can you open a new discussion with the details of the issue you're seeing since you're using a different CMS? That will help us debug more effectively.

**Reply by haseeb-bewonder** [reply]
Hi @amyegan , In fact I'm really new to vercel and don't know much about it so do you think will it be worth it to open new ticket ? Though I'm using Wordpress(a different CMS than the one in question) but the situation is pretty much same and I'm unable to find the solution
